src/services/authorization-finder.js

```diff
@@ -36,7 +37,18 @@ function AuthorizationFinder(
       })
       .catch((error) => {
         logger.error('Authorization error: ', error);
-        throw new Error();
+        let errorMessageToForward;
+        switch (error.message) {
+          case errorMessages.SERVER_TRANSACTION.SECRET_AND_RENDERINGID_INCONSISTENT:
+            errorMessageToForward = error.message;
+            break;
+          case errorMessages.SERVER_TRANSACTION.SECRET_NOT_FOUND:
+            errorMessageToForward = 'Cannot retrieve the project you\'re trying to unlock. '
+              + 'Please check that you\'re using the right environment secret regarding your project and environment.';
+            break;
+          default: errorMessageToForward = undefined;
+        }
+        throw new Error(errorMessageToForward);
```

Comment:
We want to throw a specific message here when a 404 or a 422 is received from the server.
In case of a 404 (SECRET_NOT_FOUND), I wrote a message more digest to be displayed in the toast for the end user.